linera-client/src/client_context.rs

```diff
@@ -417,6 +417,7 @@ where
         Fut: Future<Output = Result<ClientOutcome<T>, E>>,
         Error: From<E>,
     {
+        client.prepare_chain().await?;
```

Comment:
Yes: They all (potentially) create a new block, which is only possible if the chain is up-to-date. Since we are not in connected mode here (not running the node service), we do need to make sure it is updated.